### PR TITLE
LPS-182219 | Delete API Application Endpoint

### PR DESCRIPTION
**Pull request by Hijky**
Background:
Add a testcase to delete API Application Endpoint

Test Plan
CanSeeWarningMessageBeforeEndpointDeleted
Given published API Application with Endpoint created
When delete API Endpoint
Then I can see the Delete API Endpoint dialog with warning message

CanCancelAPIEndpointDeletion
Given published API Application with Endpoint created
And Given delete API Endpoint
When I click cancel button
Then I can see the API Endpoint in the UI

CanDeleteAPIEndpoint
Given published API Application with Endpoint created
And Given goto dropdown menu and click delete
When I click delete button
Then the API Endpoint has been deleted in the UI with a successful message

CannotSeeComapnyScopedEndpointInAPIExplorerAfterDelete
Given published API Application with Endpoint created
When I delete the API Endpoint
Then API Endpoint is not visible in the created API Application in API Explorer

CannotSeeSiteScopedEndpointInAPIExplorerAfterDelete
Given published API Application with group scoped Endpoint with related schema created
And Given I delete the API Endpoint
when navigate to API Explorer’s created API Application 
Then API Endpoint is not visible under {schema} in the created API Application in API Explorer

Jira ticket:
https://liferay.atlassian.net/browse/LPS-194888

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro [hunk 1]
```diff
@@ -43,10 +43,18 @@ definition {
 				"apiApplicationToAPIEndpoints": [{
 					"httpMethod": "get",
 					"name": "${endpointName}",
-					"path": "${endpointPath}"
+					"path": "${endpointPath}",
+					"scope": {"name": "${scopeKey}"}
 				}]
 			''';
 
+			if (isSet(scopeKey)) {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro:51]: If we pass`scopeKey=company`, then the codes doesn't work correctly, please update here

**Review by yanan — CHANGES_REQUESTED** [review]
Please check the upper comment, others all good!

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro (hunk 1 above)
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro:51]: ok, I will update it soon

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Review by yanan — APPROVED** [review]

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**